pages/_app.js

```diff
@@ -1,9 +1,16 @@
 import GlobalStyle from "../styles";
 import Layout from "@/components/Layout";
+import Header from "@/components/Header";
+import { useState } from "react";
 
 export default function App({ Component, pageProps }) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleToggle = () => setIsOpen(!isOpen);
+
   return (
     <>
+      <Header isOpen={isOpen} onToggle={handleToggle} />
```

Comment:
I think since you now have your header component in your Layout component you don't need it here anymore, right?